engine/src/main/java/org/hibernate/validator/constraints/UniqueItems.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraints;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+import org.hibernate.validator.internal.constraintvalidators.hv.UniqueItemsValidator;
+
+/**
+ * Validates that the provided Collection is a Set of unique items. By default, uniqueness is defined using the object's
+ * equals method.
+ * <p>
+ * This is necessary because JAX-RS always deserializes to a list. Thus, duplicates are implicitly and silently removed
+ * when converting it to a set. This annotation will allow you to check for duplicates in the list and raise an error
+ * instead.
+ *
+ * @author Tadhg Pearson
+ */
+@Documented
+@Target({ ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = UniqueItemsValidator.class)
+public @interface UniqueItems {
+
+	String message() default "{org.hibernate.validator.constraints.UniqueItems.message}";
```

Comment:
So the idea is that you should declare this string with the other constraints in ValidationMessages.properties so that it can be translated.